### DISCUSSION TITLE
Conversion MAT File to NWB file format using PyNWB

### DISCUSSION THREAD

**Discussion by imeMFK01** [NeurodataWithoutBorders/helpdesk#89] [General]
Based on [this](https://github.com/NeurodataWithoutBorders/mat2nwb/issues/1) can anyone guide me how I can convert *.mat (MATLAB) file using PyNWB because this is my stringent required approach to follow.
Thanks in advance.
@oruebel @bendichter

**Comment by bendichter** [comment]
.mat is simply a file format and these files can contain all sorts of different kinds of data. What data do your files contain? Trails? Subject information? Electrophysiology? 2p data? Knowing what sorts of data will help me direct you to the right tutorials.

**Reply by imeMFK01** [reply]
Hi, nice to hear from you.
Here is the [Link](https://sparc.science/datasets/375?type=dataset&datasetDetailsTab=files&path=files/primary/sub-DP8/perf-DP8-random-patterns) of dataset. My dataset is **TRIAL** based **ELECTROPHYSIOLOGY** data.
Please let me know if you require any further information.
Thanks.

**Reply by nmarkowitz** [reply]
If it's a matlab file then just load the file into a python session and convert your data from there. Load it using 
```
from pymatreader import read_mat
data = read_mat("my_matlab_file.mat")
```

After that you can use the [tutorials from pynwb](https://pynwb.readthedocs.io/en/stable/tutorials/index.html) to convert. Like all other files matlab files don't have a stringent structure unless they already follow a specific structure such as fieldtrip or EEGLAB

**Reply by imeMFK01** [reply]
Thanks.
